### PR TITLE
Tim/stable json

### PR DESCRIPTION
**Pull request by tux3**

### CODE REVIEW

**Review by JMounier — APPROVED** [review]

File: b64json/b64json.go [hunk 1]
```diff
@@ -3,8 +3,37 @@ package b64json
...
+func indexOf(needle string, data []string) int {
+	for k, v := range data {
+		if needle == v {
+			return k
+		}
+	}
+	return -1
+}
+
+func rubyJsonSort(a *orderedmap.Pair, b *orderedmap.Pair) bool {
```
Comment (AlexandreBossard) [inline b64json/b64json.go:18]: Just call it sort, It has nothing to do with ruby, it's to be considered arbitrary at this point

Comment (tux3) [reply b64json/b64json.go]: Sure thing.

**Review by AlexandreBossard — APPROVED** [review]

File: b64json/b64json.go [hunk 2]
```diff
@@ -3,8 +3,37 @@ package b64json
...
+		"ephemeral_public_signature_key",
+		"ephemeral_private_signature_key",
+		"user_secret",
+		"public_encryption_key",
+		"private_encryption_key",
+		"public_signature_key",
+		"private_signature_key",
+	}
+	aIdx := indexOf(a.Key(), jsonOrder)
+	bIdx := indexOf(b.Key(), jsonOrder)
```
Comment (blastrock) [inline b64json/b64json.go:33]: Why not a map like you did in other languages?

File: b64json/b64json.go [hunk 3]
```diff
@@ -13,7 +42,20 @@ func Encode(v interface{}) (*string, error) {
...
-	b64Token := base64.StdEncoding.EncodeToString(jsonToken)
+	// Struct fields are marshalled in order of declaration, but we can't easily change the order
+	// OrderedMap fields are always marshalled in order, so we bounce through it
+	o := orderedmap.New()
+	err = o.UnmarshalJSON(jsonToken)
+	if err != nil {
+		return nil, err
+	}
+	o.Sort(jsonSort)
+	orderedJson, err2 := json.Marshal(o)
```
Comment (blastrock) [inline b64json/b64json.go:53]: can't reuse `err`?

Comment (tux3) [reply b64json/b64json.go:53]: The := syntax wouldn't let me shadow err, would you like a predeclaration of orderedJson, rather?

File: b64json/b64json.go (hunk 2 above)
Comment (tux3) [reply b64json/b64json.go]: Good point, fixed.

**Review by blastrock — APPROVED** [review]

**Pull request merged**